cas-server-integration-ehcache/src/main/java/org/jasig/cas/ticket/registry/EhCacheTicketRegistry.java

```diff
@@ -50,7 +51,7 @@
  * @author Andrew Tillinghast
  * @since 3.5
  */
-public final class EhCacheTicketRegistry extends AbstractDistributedTicketRegistry implements InitializingBean {
+public final class EhCacheTicketRegistry extends AbstractCrypticTicketRegistry implements InitializingBean {

```

Comment:
Re-reading and reviewing everything once again, I understand things better: the new `AbstractCrypticTicketRegistry` only applies to distributed ticket registries. So you just need to set a cypher executor to add data protection and it does not break backward compatibility.

Yes, it's a better solution than my proposal.